### ISSUE TITLE
Bug: v7.32 reports "react/jsx-key" for all children of a <div>

### ISSUE THREAD

**Issue by rally25rs** [eslint/eslint#15645]
### Environment

Node version: v16.14.0
npm version: yarn 3.0.2
Local ESLint version: 7.32.0
Global ESLint version: none
Operating System: OSX

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports = {
  plugins: ['react-hooks'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: '2018',
    project: './tsconfig.json',
    sourceType: 'module',
  },
  ignorePatterns: [
    '.eslintrc.js',
    '.yarn/',
  ],
  settings: {
    react: {
      version: '17.0',
    },
  },
  extends: [
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
  ],
  env: {
    browser: true,
    es6: true,
    node: true,
  },
  overrides: [
    // typescript
    {
      files: ['*.ts', '*.tsx'],
      excludedFiles: ['*.js'],
      plugins: ['@typescript-eslint', 'react-hooks'],
      extends: [
        'plugin:@typescript-eslint/recommended',
        'plugin:react/recommended',
        'plugin:react/jsx-runtime',
      ],
    },
  ],
};
```
</details>

```js
export const ExampleTwo = () => {
  return (
    <div>
      <button type="button">Clear</button>
    </div>
  );
};
```

### What did you expect to happen?

No linting errors

### What actually happened?

```
$ yarn eslint packages/react-components/src/test.stories.tsx

/Users/me/Projects/eslint-test/packages/react-components/src/test.stories.tsx
  4:7  error  Missing "key" prop for element in array  react/jsx-key

✖ 1 problem (1 error, 0 warnings)
```

eslint reports all child elements of a `<div>` as needing a "key" prop.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Added a minimal repro example : https://github.com/rally25rs/eslint-issue-15645

**Comment by mdjermanovic** [comment]
Hi @rally25rs, thanks for the issue!

> 4:7  error  Missing "key" prop for element in array  react/jsx-key

This error is coming from a plugin rule [`react/jsx-key`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-key.md), which isn't maintained in this repository.

Can you report the issue in the plugin's repository:

https://github.com/yannickcr/eslint-plugin-react

I'm closing this issue since it isn't related to core ESLint or core ESLint rules.
